blockbase/functions.php

```diff
@@ -187,3 +134,99 @@ function blockbase_core_navigation_render_fallback() {
 if ( file_exists( get_stylesheet_directory() . '/inc/block-patterns.php' ) ) {
 	require_once get_stylesheet_directory() . '/inc/block-patterns.php';
 }
+
+
+// ----------	Custom Fonts ---------- //
+const BLOCKBASE_GOOGLE_FONTS_LIST = array(
+	'Arvo',
+	'Bodoni Moda',
+	'Cabin',
+	'Chivo',
+	'Courier Prime',
+	'DM Sans',
+	'Domine',
+	'EB Garamond',
+	'Fira Sans',
+	'IBM Plex Sans',
+	'IBM Plex Mono',
+	'Inter',
+	'Josefin Sans',
+	'Jost',
+	'Libre Baskerville',
+	'Libre Franklin',
+	'Literata',
+	'Lora',
+	'Merriweather',
+	'Montserrat',
+	'Newsreader',
+	'Nunito',
+	'Open Sans',
+	'Overpass',
+	'Playfair Display',
+	'Poppins',
+	'Raleway',
+	'Red Hat Display',
+	'Roboto',
+	'Roboto Slab',
+	'Rubik',
+	'Source Sans Pro',
+	'Source Serif Pro',
+	'Space Mono',
+	'Texturina',
+	'Work Sans',
+);
+
+/**
+ * Register a curated selection of Google Fonts.
+ *
+ * @return void
+ */
+function blockbase_register_google_fonts() {
+	// Use jetpack's implementation of custom google fonts if it is already active
+	if ( method_exists( 'Jetpack', 'is_module_active' ) && Jetpack::is_module_active( 'google-fonts' ) ) {
+		return;
+	} 
+
+	if ( ! function_exists( 'wp_register_webfont_provider' ) || ! function_exists( 'wp_register_webfonts' ) ) {
+		return;
+	}
+
+	wp_register_webfont_provider( 'blockbase-google-fonts', '\Automattic\Jetpack\Fonts\Google_Fonts_Provider' );
+
+	/**
+	 * Curated list of Google Fonts.
+	 *
+	 * @module google-fonts
+	 *
+	 * @since 10.8
+	 *
+	 * @param array $fonts_to_register Array of Google Font names to register.
+	 */
+	$fonts_to_register = apply_filters( 'blockbase_google_fonts_list', BLOCKBASE_GOOGLE_FONTS_LIST );
+
+	foreach ( $fonts_to_register as $font_family ) {
+		wp_register_webfonts(
+			array(
+				array(
+					'font-family'  => $font_family,
+					'font-weight'  => '100 900',
+					'font-style'   => 'normal',
+					'font-display' => 'swap',
+					'provider'     => 'blockbase-google-fonts',
+				),
+				array(
+					'font-family'  => $font_family,
+					'font-weight'  => '100 900',
+					'font-style'   => 'italic',
+					'font-display' => 'swap',
+					'provider'     => 'blockbase-google-fonts',
+				),
+			)
```

Comment:
I think we actually have the opportunity to be more selective here with the font weights by including only those needed by the themes, reducing the number of font-face rules that are output on the page.

What do you think of modifying `BLOCKBASE_GOOGLE_FONTS_LIST` to include the `google` information from `GlobalStylesFontsCustomizer->$fonts`, and load only the weights specified for each font?

Reply:
(I'm wondering if we should do something similar with the Google Fonts module, as well 🤔)

Reply:
> What do you think of modifying BLOCKBASE_GOOGLE_FONTS_LIST to include the google information from GlobalStylesFontsCustomizer->$fonts, and load only the weights specified for each font?

I think this is fair, but would we mind doing it in a follow-up, seeing as how this is the same behavior that's already implemented in production for the Jetpack Google Fonts Module? That way we can tackle it in Blockbase 
_and_ Jetpack at the same time. Happy to create a ticket for us if you're okay with it. 👍

Reply:
From what I can tell the above is including only a single weight per-font family.  I'm not sure I see how more selective it can be.  Or perhaps I'm reading it wrong...

Reply:
> I think this is fair, but would we mind doing it in a follow-up, seeing as how this is the same behavior that's already implemented in production for the Jetpack Google Fonts Module?

That's fine by me!

Reply:
https://github.com/Automattic/wp-calypso/issues/64303